src/Storages/ExternalStream/ExternalStreamSettings.h

```diff
@@ -43,6 +45,25 @@ DECLARE_SETTINGS_TRAITS(ExternalStreamSettingsTraits, LIST_OF_EXTERNAL_STREAM_SE
 struct ExternalStreamSettings : public BaseSettings<ExternalStreamSettingsTraits>
 {
     void loadFromQuery(ASTStorage & storage_def);
+
+    FormatSettings getFormatSettings(const ContextPtr & context)
+    {
+        FormatFactorySettings settings {};
+        const auto & settings_from_context = context->getSettingsRef();
+
+        /// settings from context have higher priority
+#define SET_CHANGED_SETTINGS(TYPE, NAME, DEFAULT, DESCRIPTION, FLAGS) \
+        if ((NAME).changed) \
+            settings.NAME = (NAME); \
+        if (settings_from_context.NAME.changed) \
```

Comment:
```
if (settings_from_context.NAME.changed)
 ...
else if ((NAME).changed)
...
```

Reply:
or 
```c++
auto settings = *this; /// clone created settings

...
if (settings_from_context.NAME.changed) /// apply changed setting from context
   ....
```

Reply:
This can be improved, as above

Reply:
Since `this` has more fields than `FormatFactorySettings` (and `this` could grow even bigger if more external streams implemented), thus I used the first one as you suggested.